wsd/DocumentBroker.cpp

```diff
@@ -1135,18 +1142,30 @@ bool DocumentBroker::download(
     }
     else
     {
-        // Check if document has been modified by some external action
+        // Check if document has been modified by some external action,
+        // but only if *we* aren't uploading. Otherwise, it might be us.
         LOG_TRC("Document modified time: " << fileInfo.getLastModifiedTime());
         if (!_storageManager.getLastModifiedTime().empty() &&
             !fileInfo.getLastModifiedTime().empty() &&
             _storageManager.getLastModifiedTime() != fileInfo.getLastModifiedTime())
         {
-            LOG_DBG("Document [" << _docKey << "] has been modified behind our back. "
-                                 << "Informing all clients. Expected: "
-                                 << _storageManager.getLastModifiedTime()
-                                 << ", Actual: " << fileInfo.getLastModifiedTime());
+            if (_uploadRequest)
+            {
+                LOG_DBG("Document [" << _docKey
+                                     << "] has been uploaded by us while downloading it, changing "
```

Comment:
message is a bit odd - there isn't an upload while downloading surely (?) this was just a CheckFileInfo running in parallel with an up-load (?) how about:

"timestamp checked for a match during an up-load, results may race, so ignoring inconsistency here" or somesuch =)